.ci/static-checks.sh

```diff
@@ -838,6 +839,58 @@ static_check_vendor()
 	dep ensure -no-vendor -dry-run
 }
 
+static_check_xml()
+{
+	local all_xml
+	local files
+
+	all_xml=$(find . -name "*.xml" | grep -v "vendor/" | sort || true)
+
+	if [ "$specific_branch" = "true" ]
+	then
+		info "Checking all XML files in $branch branch"
+		files="$all_xml"
+	else
+		info "Checking local branch for changed XML files only"
+
+		local xml_status
+
+		xml_status=$(get_pr_changed_file_details || true)
+		xml_status=$(echo "$xml_status" | grep "\.xml$" || true)
+
+		files=$(echo "$xml_status" | awk '{print $NF}')
+	fi
+
+	[ -z "$files" ] && info "No XML files to check" && return
+
+	local file
+
+	for file in $files
+	do
+		info "Checking XML file '$file'"
+
+		local contents
+
+		# Most XML documents are specified as XML 1.0 since, with the
+		# advent of XML 1.0 (Fifth Edition), XML 1.1 is "almost
+		# redundant" due to XML 1.0 providing the majority of XML 1.1
+		# features. xmllint doesn't support XML 1.1 seemingly for this
+		# reason, so the only check we can do is to (crudely) force
+		# the document to be an XML 1.0 one since XML 1.1 documents
+		# can mostly be represented as XML 1.0.
+		#
+		# This is only really required since Jenkins creates XML 1.1
```

Comment:
Ah, so, this is with the (at least partial) intent of checking the Jenkins xml files... which, whilst good, if one of them is *not* valid XML, I'm not sure there is a huge amount we could do about it - the files are generated by Jenkins when we modify the GUI input - unclear if we would/could then go and hand-fix the xml and fold it back to the Jenkins server...

Reply:
Although most of the XML files are Jenkins, there are other XML files in our repos. Regardless of who or what generated the files, we *really* do need to know if the files are valid or not. Particularly if you consider the fact that Jenkins is a core infrastructure tool and add a hypothetical disaster recovery scenario into the mix... :smile: